crates/ruff_server/src/session/index.rs

```diff
@@ -338,22 +352,38 @@ impl Index {
         &mut self,
         key: &DocumentKey,
     ) -> crate::Result<&mut DocumentController> {
-        let Some(path) = self.path_for_key(key).cloned() else {
+        let Some(url) = self.url_for_key(key).cloned() else {
             anyhow::bail!("Tried to open unavailable document `{key}`");
         };
-        let Some(controller) = self.documents.get_mut(&path) else {
-            anyhow::bail!("Document controller not available at `{}`", path.display());
+        let Some(controller) = self.documents.get_mut(&url) else {
+            anyhow::bail!("Document controller not available at `{}`", url);
         };
         Ok(controller)
     }
 
-    fn path_for_key<'a>(&'a self, key: &'a DocumentKey) -> Option<&'a PathBuf> {
+    fn url_for_key<'a>(&'a self, key: &'a DocumentKey) -> Option<&'a Url> {
         match key {
             DocumentKey::Notebook(path) | DocumentKey::Text(path) => Some(path),
             DocumentKey::NotebookCell(uri) => self.notebook_cells.get(uri),
         }
     }
 
+    fn settings_for_url(&self, url: &Url) -> Option<&WorkspaceSettings> {
+        if let Some(path) = url.to_file_path().ok() {
+            self.settings_for_path(&path)
+        } else {
+            // If there's only a single workspace, use that configuration for an untitled document.
+            if self.settings.len() == 1 {
+                tracing::debug!(
+                    "Falling back to configuration of the only active workspace for the new document '{url}'."
+                );
+                self.settings.values().next()
+            } else {
+                None
+            }
```

Comment:
I'm not entirely sure about this. It's nice because the file is most likely to be part of the workspace and it would be unfortunate if we show lint errors for rules that are disabled in this workspace. 

However, it has the downside that the shown lint errors change when the user has an unsaved document and then opens a new workspace. A next pull diagnostic now uses the default configuration which is different from the last run. I think that's fine, considering that adding a workspace AND having an unsaved file is rare.